.github/workflows/build-cloudberry.yml

```diff
@@ -17,102 +17,86 @@
 # permissions and limitations under the License.
 #
 # --------------------------------------------------------------------
+# GitHub Actions Workflow: Apache Cloudberry Build Pipeline
+# --------------------------------------------------------------------
+# Description:
 #
-# GitHub Actions Workflow: Apache Cloudberry Build
-# Description: Builds, tests, and packages Apache Cloudberry on Rocky Linux 9
+#   This workflow builds, tests, and packages Apache Cloudberry on
+#   Rocky Linux 9. It ensures artifact integrity, performs installation
+#   tests, validates key operations, and provides detailed test reports,
+#   including handling for ignored test cases.
 #
-# Workflow Structure:
-# 1. Skip Check:
-#    - Evaluates CI skip flags in commit messages or PR titles
-#    - Supports [skip ci], [ci skip], [no ci]
-#    - Successfully completes all checks if skip detected
+# Workflow Overview:
+# 1. **Check Skip**:
+#    - Dynamically determines if the workflow should run based on CI skip flags.
+#    - Evaluates the following fields for skip flags:
+#      - **Pull Request Events**: PR title and PR body.
+#      - **Push Events**: Commit message of the head commit.
+#    - Supports the following skip patterns (case-insensitive):
+#      - `[skip ci]`
```

Comment:
Thanks for supporting this cool feature. One question, code modification and final merging require running through CI at least once, but submitting code during review can skip CI, right?

Reply:
> Thanks for supporting this cool feature. One question, code modification and final merging require running through CI at least once, but submitting code during review can skip CI, right?

@my-ship-it - Yes, that's correct. For changes that don't affect the Cloudberry source code (like documentation updates), you can skip the CI pipeline by adding any of these tags to your PR title or body:

```
[skip ci]
[ci skip]
[no ci]
```

However, as you noted, code changes and merging will still require passing through CI for quality assurance. This feature will be particularly useful for our upcoming Apache incubation-related PRs, allowing us to move more efficiently with non-code changes.